src/deploy/functions/release/fabricator.ts

```diff
@@ -606,3 +647,38 @@ export class Fabricator {
     utils.logSuccess(`${clc.bold.green(`functions[${label}]`)} Successful ${op} operation.`);
   }
 }
+
+/**
+ * Returns whether a service is resolved (all transitions have completed).
+ */
+export function serviceIsResolved(service: run.Service): boolean {
+  if (service.status?.observedGeneration !== service.metadata.generation) {
+    logger.debug(
+      `Service ${service.metadata.name} is not resolved because` +
+        `observed generation ${service.status?.observedGeneration} does not ` +
+        `match spec generation ${service.metadata.generation}`
+    );
+    return false;
+  }
+  const readyCondition = service.status?.conditions?.find((condition) => {
+    return condition.type === "Ready";
+  });
+
+  if (readyCondition?.status === "Unknown") {
+    logger.debug(
+      `Waiting for service ${service.metadata.name} to be ready. ` +
+        `Status is ${JSON.stringify(service.status?.conditions)}`
+    );
+    return false;
+  } else if (readyCondition?.status === "True") {
+    return true;
+  }
+  logger.debug(
+    `Service ${service.metadata.name} has unexpected ready status ${JSON.stringify(
```

Comment:
```suggestion
    `Service ${service.metadata.name} has unexpected status ${JSON.stringify(
```

Reply:
We're specifically talking about the Status of Type=Ready so I think the comment is correct.